ui/src/ducks/inventory/index.js

```diff
@@ -1,17 +1,91 @@
+import axios from 'axios'
 import { createAction, handleActions } from 'redux-actions'
+import { openSuccess } from '../alerts'
 
 const actions = {
   INVENTORY_GET_ALL: 'inventory/get_all',
+  INVENTORY_PENDING: 'inventory/pending',
+  INVENTORY_CREATE: 'inventory/create',
+  INVENTORY_UPDATE: 'inventory/update',
+  INVENTORY_DELETE: 'inventory/delete',
+  INVENTORY_REFRESH: 'inventory/refresh'
 }
 
 export let defaultState = {
-  all: []
+  all: [],
+  fetched: false,
 }
 
-export const findInventory = createAction(actions.INVENTORY_GET_ALL, () => {
-  //TODO
-})
+export const findInventory = createAction(actions.INVENTORY_GET_ALL, () =>
+  (dispatch, getState, config) => axios
+    .get(`${config.restAPIURL}/inventory`)
+    .then((suc) => {
+      openSuccess()
+      dispatch(refreshInventory(suc.data))
+  })
+)
+export const createInventory = createAction(actions.INVENTORY_CREATE, (inventory) => 
+  (dispatch, getState, config) => axios
+    .post(`${config.restAPIURL}/inventory`, inventory)
+    .then((suc) => {
+      openSuccess()
+      const invs = []
+      //adds all the inventory not matching the created one to the array
+      getState().inventories.all.forEach(inv => {
+        if (inv.id !== suc.data.id) {
+          invs.push(inv)
+        }
+      })
+      //adds created inventory to the array
+      invs.push(suc.data)
+      //refresh with contents of the array
+      dispatch(refreshInventory(invs))
+    })
+)
+export const deleteInventory = createAction(actions.INVENTORY_DELETE, (id) => 
+  (dispatch, getState, config) => axios
+    .delete(`${config.restAPIUrl}/inventory`, {data: id})
+    .then((suc) => {
+      openSuccess()
+      const invs = []
+      getState().inventories.all.forEach(inv => {
+        //pushes all the inventories that don't include the deleted id
+        if (!id.includes(inv.id)) {
+          invs.push(inv)
+        }
+      })
+      dispatch(refreshInventory(invs))
+    })
+    
+)
+export const updateInventory = createAction(actions.INVENTORY_UPDATE, (id, inventory) => 
+  (dispatch, getState, config) => axios
+    .put(`${config.restAPIUrl}/inventory/${id}`,inventory)
+    .then((suc) => {
+      openSuccess()
+      const invs = []
+      getState().inventories.all.forEach(inv => {
+        if (inv.id !== suc.data.id) {
+          invs.push(inv)
+        }
+      })
+      invs.push(suc.data)
+      dispatch(refreshInventory(invs))
+    })
+)
+export const refreshInventory = createAction(actions.INVENTORY_REFRESH, (payload) =>
+  (dispatcher, getState, config) =>
+    payload.sort((inventoryA, inventoryB) => inventoryA.name <inventoryB.name ? -1 : inventoryA.name > inventoryB.name ? 1: 0)
+)
 
 export default handleActions({
-  //TODO
+  [actions.INVENTORY_PENDING]: (state) => ({
```

Comment:
While the "_PENDING" action is not used at the moment, it's important later on to keep track of the front end state (e.g., should a component render when `fetched` is true while the backend is being retrieved?).

"INVENTORY_PENDING" isn't tracking any function's state; make it similar to `product/index.js` by tracking "INVENTORY_GET_ALL".